tests/task.test.ts

```diff
@@ -12,135 +12,223 @@ const index = {
-      expect(stausReponse).toHaveProperty('uid', expect.any(Number))
-      expect(stausReponse).toHaveProperty('type', 'documentAddition')
-      expect(stausReponse).toHaveProperty('details')
-      expect(stausReponse.details).toHaveProperty('indexedDocuments', 7)
-      expect(stausReponse.details).toHaveProperty('receivedDocuments', 7)
-      expect(stausReponse).toHaveProperty('duration', expect.any(String))
-      expect(stausReponse).toHaveProperty('enqueuedAt', expect.any(String))
-      expect(stausReponse).toHaveProperty('finishedAt', expect.any(String))
-      expect(stausReponse).toHaveProperty('startedAt', expect.any(String))
+
+      const enqueuedTask = await client.index(index.uid).addDocuments(dataset)
+
+      expect(enqueuedTask.taskUid).toBeDefined()
+      expect(enqueuedTask.indexUid).toEqual(index.uid)
+      expect(enqueuedTask.status).toBeDefined()
+      expect(enqueuedTask.type).toEqual(TaskTypes.DOCUMENTS_ADDITION_OR_UPDATE)
+      expect(enqueuedTask.enqueuedAt).toBeDefined()
     })
 
-    test(`${permission} key: Get one update`, async () => {
+    test(`${permission} key: Get one task`, async () => {
       const client = await getClient(permission)
-      const response: EnqueuedTask = await client
-        .index(index.uid)
-        .addDocuments(dataset)
-      expect(response).toHaveProperty('uid', expect.any(Number))
-      await client.waitForTask(response.uid)
+      const enqueuedTask = await client.index(index.uid).addDocuments(dataset)
+      await client.waitForTask(enqueuedTask.taskUid)
 
-      const stausReponse: Task = await client
-        .index(index.uid)
-        .getTask(response.uid)
-
-      expect(stausReponse).toHaveProperty('status', TaskStatus.TASK_SUCCEEDED)
-      expect(stausReponse).toHaveProperty('uid', expect.any(Number))
-      expect(stausReponse).toHaveProperty('type', 'documentAddition')
-      expect(stausReponse).toHaveProperty('details')
-      expect(stausReponse.details).toHaveProperty('indexedDocuments', 7)
-      expect(stausReponse.details).toHaveProperty('receivedDocuments', 7)
-      expect(stausReponse).toHaveProperty('duration', expect.any(String))
-      expect(stausReponse).toHaveProperty('enqueuedAt', expect.any(String))
-      expect(stausReponse).toHaveProperty('finishedAt', expect.any(String))
-      expect(stausReponse).toHaveProperty('startedAt', expect.any(String))
+      const task = await client.getTask(enqueuedTask.taskUid)
+
+      expect(task.indexUid).toEqual(index.uid)
+      expect(task.status).toEqual(TaskStatus.TASK_SUCCEEDED)
+      expect(task.type).toEqual(TaskTypes.DOCUMENTS_ADDITION_OR_UPDATE)
+      expect(task.enqueuedAt).toBeDefined()
+      expect(task.uid).toEqual(enqueuedTask.taskUid)
+      expect(task).toHaveProperty('details')
+      expect(task.details.indexedDocuments).toEqual(7)
+      expect(task.details.receivedDocuments).toEqual(7)
+      expect(task.duration).toBeDefined()
```

Comment:
The type now is protected by the typescript, don't need to be covered again the tests, right?

Reply:
I'm not sure which tests you're suggesting that I remove? They are also kind of testing Meilisearch.